packages/components/data-table-manager/src/data-table-manager.tsx

```diff
@@ -166,7 +166,10 @@ const DataTableManager = <Row extends TRow = TRow>(
     [areDisplaySettingsEnabled, props.columns, isWrappingText]
   );
 
-  return (
+  //  @ts-ignore
+  window.dataTableColumns = { columns };
```

Comment:
I would use the window.dispatch but it seems like we would be missing the initial state using this approach
```suggestion
  useEffect(()=> {
    const customEvent = new CustomEvent(‘DataTaBleManagerEvent’, {
      detail: {columns}
    })
    window.dispatchEvent(customEvent)
  },[columns])
```

Reply:
Thank you for the comprehensive feedback. I will push a change with the Provider approach since it is now clear that we would have some security/complexity concerns using the window approach 🙏🏾

Reply:
updated here: https://github.com/commercetools/ui-kit/pull/2807/commits/39a97a46ad4d52bc58293c40b7f07325c8ad5a55

Thank you.